x-pack/test/detection_engine_api_integration/security_and_spaces/tests/create_exceptions.ts

```diff
@@ -394,6 +396,83 @@ export default ({ getService }: FtrProviderContext) => {
         ]);
       });
 
+      it('should not change the immutable tags when adding a second exception list to an immutable rule through patch', async () => {
+        await installPrePackagedRules(supertest);
+
+        const { id, list_id, namespace_type, type } = await createExceptionList(
+          supertest,
+          getCreateExceptionListMinimalSchemaMock()
+        );
+
+        // Rule id of "9a1a2dae-0b5f-4c3d-8305-a268d404c306" is from the file:
+        // x-pack/plugins/security_solution/server/lib/detection_engine/rules/prepackaged_rules/elastic_endpoint.json
+        // This rule has an existing exceptions_list that we are going to use
+        const immutableRule = await getRule(supertest, '9a1a2dae-0b5f-4c3d-8305-a268d404c306');
+        expect(immutableRule.exceptions_list.length).greaterThan(0); // make sure we have at least one
+
+        // add a second exceptions list as a user is allowed to add a second list to an immutable rule
+        await supertest
+          .patch(DETECTION_ENGINE_RULES_URL)
+          .set('kbn-xsrf', 'true')
+          .send({
+            rule_id: '9a1a2dae-0b5f-4c3d-8305-a268d404c306',
+            exceptions_list: [
+              ...immutableRule.exceptions_list,
+              {
+                id,
+                list_id,
+                namespace_type,
+                type,
+              },
+            ],
+          })
+          .expect(200);
+
+        const body = await findImmutableRuleById(supertest, '9a1a2dae-0b5f-4c3d-8305-a268d404c306');
+        expect(body.data.length).to.eql(1); // should have only one length to the data set, otherwise we have duplicates or the tags were removed and that is incredibly bad.
+
+        const bodyToCompare = removeServerGeneratedProperties(body.data[0]);
+        expect(bodyToCompare.rule_id).to.eql(immutableRule.rule_id); // Rule id should not change with a a patch
+        expect(bodyToCompare.immutable).to.eql(immutableRule.immutable); // Immutable should always stay the same which is true and never flip to false.
+        expect(bodyToCompare.version).to.eql(immutableRule.version); // The version should never update on a patch
```

Comment:
I know this is expected, but I wonder if we do want to be updating versions to some extent for exceptions (not when adding an exception item, but adding or removing a list) since it does greatly change results.

Reply:
That's a good question. We update the version number when we add/change/remove actions fwiw on mutable rules. For this test if it helps out we should not be changing version when it comes to immutable rules since we need the field for updates to immutables and to know when they are different on updates.

Reply:
Just checked and for mutable rules it looks like the version number is bumped when we add an exception list so I think we're good here. Above this, I think there might be tests to verify that behavior.

Reply:
Ah ok, thanks for testing that out!